### PR TITLE
Microgateway APIM Internal-Key Support

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
### Purpose
This PR will add the internalKeyAuthenticator to mgw in order to support APIM publisher test.
Step 1. Generate Internal Key at APIM publisher
1. Generate token with to api_generate_key scope and other required scopes to access the publisher
curl -k -X POST https://localhost:9443/oauth2/token -d 
"grant_type=password&username=username&password=password&scope=apim:api_view apim:api_create apim:api_generate_key" -H "Authorization: Basic consumerKey:consumerSecret"

2. Add new API, get the UUID of that API and call generate-key endpoint using the above step1 token
curl -k -X POST -H "Authorization: Bearer token" -H "Content-Type: application/json" "https://127.0.0.1:9443/api/am/publisher/v2/apis/uuid/generate-key"

3. Send the request with adding Internal-Key authentication header
Eg:-
curl -X GET "https://localhost:9095/pizzashack/1.0.0/menu" -H "accept: application/json" -H "Internal-Key: $APIKey" -k

### Issues
https://github.com/wso2/product-microgateway/issues/1746

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/InternalAPIKeyAuthenticator.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+        if (requestContext.getMathedAPI() != null) {
+            if (log.isDebugEnabled()) {
+                log.info("Internal Key Authentication initialized");
+            }
+
+            try {
+                // Extract internal from the request while removing it from the msg context.
+                String internalKey = extractInternalKey(requestContext);
+                if (StringUtils.isEmpty(internalKey)) {
+                    log.error("Internal Key does not perform at request");
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/InternalAPIKeyAuthenticator.java:76]: Shall we make this message more meaningful

Comment (HiranyaKavishani) [reply enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/InternalAPIKeyAuthenticator.java:76]: +1

Comment (HiranyaKavishani) [reply enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/InternalAPIKeyAuthenticator.java]: Fixed by https://github.com/wso2/product-microgateway/pull/1747/commits/18efbb14f8e3d8b39eac836bf490639e3c126d6f

File: adapter/internal/api/apis_impl.go [hunk 2]
```diff
@@ -121,20 +124,14 @@ func extractAPIProject(payload []byte) (apiProject ProjectAPI, err error) {
 				return apiProject, conversionErr
 			}
 
-			apiType, lifeCycleStatus, productionEndpoint, sandboxEndpoint, endpointImplementationType, err :=
-				extractAPIInformation(apiJsn)
-
-			if endpointImplementationType == "INLINE" {
+			apiProject.APIJsn = apiJsn
+			extractAPIInformation(apiProject)
+			if apiProject.endpointImplementationType == "INLINE" {
```
Comment (dinusha92) [inline adapter/internal/api/apis_impl.go:129]: Shall we move "INLINE" to constant?

Comment (HiranyaKavishani) [reply adapter/internal/api/apis_impl.go]: Fixed by https://github.com/wso2/product-microgateway/pull/1747/commits/1b3cf6499331400c8f4545af0f444f99077c971f

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**